src/starkware_utils/commitment_tree/merkle_tree/traverse_tree.rs

```diff
@@ -0,0 +1,48 @@
+use std::collections::VecDeque;
+
+use crate::starkware_utils::commitment_tree::errors::TreeError;
+use crate::starkware_utils::commitment_tree::leaf_fact::LeafFact;
+use crate::storage::storage::{HashFunctionType, Storage};
+
+pub trait TreeTraverser<'test, S, H, LF>
```

Comment:
why `'test`?

Reply:
An artefact from experiences with async + recursion + lifetimes, I'll rename it.